implementations/elixir/ockam/ockam_kafka/lib/interceptor/inlet_manager.ex

```diff
@@ -0,0 +1,127 @@
+defmodule Ockam.Kafka.Interceptor.InletManager do
+  @moduledoc """
+  Dynamic inlet manager for kafka interceptor.
+
+  Inlets are GenServers and cannot be managed by ockam registry.
+  This module can dynamically create and stop inlets using base port and port offset.
+  """
+  use GenServer
+
+  def start_link([base_port, allowed_ports, base_route, outlet_prefix]) do
+    GenServer.start_link(
+      __MODULE__,
+      [base_port, allowed_ports, base_route, outlet_prefix],
+      ## TODO: make this optional
+      name: __MODULE__
+    )
+  end
+
+  @impl true
+  def init([base_port, allowed_ports, base_route, outlet_prefix]) do
+    {:ok,
+     %{
+       base_port: base_port,
+       allowed_ports: allowed_ports,
+       base_route: base_route,
+       outlet_prefix: outlet_prefix,
+       inlets: %{}
+     }}
+  end
+
+  def list_inlets(server \\ __MODULE__, timeout \\ 5000) do
+    GenServer.call(server, :list_inlets, timeout)
+  end
+
+  def set_inlets(server \\ __MODULE__, port_offsets, timeout \\ 5000)
+      when is_list(port_offsets) do
+    GenServer.call(server, {:set_inlets, port_offsets}, timeout)
+  end
+
+  @impl true
+  def handle_call(:list_inlets, _from, %{inlets: inlets} = state) do
+    {:reply, inlets, state}
+  end
+
+  def handle_call(
+        {:set_inlets, requested_port_offsets},
+        _from,
+        %{allowed_ports: allowed_ports, inlets: inlets} = state
+      ) do
+    case Enum.any?(requested_port_offsets, fn port_offset ->
+           port_offset > allowed_ports or port_offset < 0
```

Comment:
Mentioned in this [comment](https://github.com/build-trust/ockam/pull/4118#discussion_r1093495527)

I think this condition should be `port_offset >= allowed_ports or port_offset < 0`, otherwise we will allow `allowed_ports + 1` ports